### PR TITLE
student view

### PR DESCRIPTION
**Pull request by Li-Evelyn**
* file upload/download - **PNGs only**
* file preview
* questionsubmission creation
* deadline submission gating
* ui for grade/staff comment on graded questionsubmissions
* small css fixes here and there

if #12 doesn't get merged in before you review this feel free to test it by logging in as user 116 and running the following commands in psql
image

that user will have one submitted/graded assignment and one new assignment, so the former is good for showing the graded ui changes and the latter is for you to submit stuff

### CODE REVIEW

File: server/query.js [hunk 1]
```diff
@@ -126,11 +126,11 @@ const Query = {
...
         await query(req, res, 'SELECT * FROM question WHERE assignment_id=$1', [aid])
     },
-    async createQuestionSubmission(req, res, uid, aid, num) {
-        await query(req, res, 'INSERT into questionsubmission (student_id, assignment_id, question_number) values ($1, $2, $3)', [uid, aid, num])
+    async getAssignmentSubmission(req, res, id, aid) {
+        await query(req, res, "SELECT * FROM assignmentsubmission WHERE student_id=$1 AND assignment_id=$2", [id, aid])
     },
-    async getAssignmentSubmissions(req, res, id) {
-        await query(req, res, 'SELECT * FROM assignmentsubmission WHERE student_id=$1', [id])
+    async getAssignmentSubmissions(req, res, id, aid=null) {
```
Comment (FizzesShark) [inline server/query.js:132]: Do we intend on using the aid parameter here?

**Review by majesticwhales — APPROVED** [review]
lgtm!

File: client/src/components/dashboard.js [hunk 2]
```diff
@@ -73,9 +73,10 @@ function Dashboard(props) {
 
     let handleAssignmentSelect = (a) => {
         setActiveAssignment(a);
+        let cname = localStorage.getItem("course_name")
```
Comment (FizzesShark) [inline client/src/components/dashboard.js:76]: This should've been fixed by my PR already

File: server/query.js (hunk 1 above)
Comment (Li-Evelyn) [reply server/query.js:132]: no, relic from an old thing

File: client/src/components/dashboard.js (hunk 2 above)
Comment (Li-Evelyn) [reply client/src/components/dashboard.js:76]: yeah that's fine, i'll update this

**Pull request merged**